src/app/core/models/profile.model.ts

```diff
@@ -0,0 +1,23 @@
+import { Schema, isValidObject } from '../../shared/lib/validate';
+
+/**
+ * Indicates all the elements that make up a Profile.
+ */
+export interface Profile {
+  profileName: string;
+  encodedText: string;
+}
+
+/**
+ * Schema for validating profiles.json entries
+ *
+ * Profile must have a profileName and encodedText field,
+ * both of which cannot be empty strings.
+ */
+
+const profileSchema: Schema = {
+  profileName: { required: true, validate: (value) => !!value },
+  encodedText: { required: true, validate: (value) => !!value }
```

Comment:
Should we have the extra layer of validation to make sure that encoded text follows the `org/repo` format? So the validate function can use regex or any form of string manipulation to verify the encoded text.

Reply:
Yup that's a good point. i guess we'll be matching for `string/string`?

Reply:
Yep! 👍